php/class-relate.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Relate class for the Cloudinary plugin.
+ *
+ * @package Cloudinary
+ */
+
+namespace Cloudinary;
+
+use Cloudinary\Connect\Api;
+use Cloudinary\Relate\Relationship;
+use WP_Query;
+
+/**
+ * Class Relate
+ */
+class Relate {
+
+	/**
+	 * Holds the plugin instance.
+	 *
+	 * @var Plugin Instance of the global plugin.
+	 */
+	public $plugin;
+
+	/**
+	 * Holds the Media instance.
+	 *
+	 * @var Media
+	 */
+	protected $media;
+
+	/**
+	 * Relate constructor.
+	 *
+	 * @param Plugin $plugin Instance of the main plugin.
+	 */
+	public function __construct( $plugin ) {
+		$this->plugin = $plugin;
+		$this->media  = $plugin->get_component( 'media' );
+		$this->register_hooks();
+	}
+
+	/**
+	 * Register hooks.
+	 */
+	protected function register_hooks() {
+		add_action( 'cloudinary_upgrade_asset', array( $this, 'upgrade_relation' ), 10, 2 );
+		add_action( 'found_posts', array( $this, 'warm_cache' ), 10, 2 );
+	}
+
+	/**
+	 * Warm the cache for found posts.
+	 *
+	 * @param int      $found_posts The number of posts found.
+	 * @param WP_Query $query       The WP_Query instance (passed by reference).
+	 */
+	public function warm_cache( $found_posts, $query ) {
+		if ( ! empty( $found_posts ) && 'attachment' === $query->query_vars['post_type'] ) {
+			Relationship::preload( $query->posts );
+		}
+	}
+
+	/**
+	 * Upgrade an asset relation.
+	 *
+	 * @param int    $attachment_id The attachment ID.
+	 * @param string $version       The version upgrading to.
+	 */
+	public function upgrade_relation( $attachment_id, $version ) {
+		$asset_plugin_version = $this->media->get_post_meta( $attachment_id, Sync::META_KEYS['plugin_version'], true );
+		if ( ! empty( $asset_plugin_version ) && version_compare( $asset_plugin_version, $version, '<' ) ) {
+			$transformations = $this->media->get_post_meta( $attachment_id, Sync::META_KEYS['transformation'], true );
+			if ( ! empty( $transformations ) ) {
+				self::update_transformations( $attachment_id, $transformations );
+			}
+			$this->media->delete_post_meta( $attachment_id, Sync::META_KEYS['transformation'] );
+		}
+	}
+
+	/**
+	 * Update transformations for an asset.
+	 *
+	 * @param int          $attachment_id   The attachment ID.
+	 * @param array|string $transformations The transformations.
+	 */
+	public static function update_transformations( $attachment_id, $transformations ) {
+		$relationship = Relationship::get_relationship( $attachment_id );
+		if ( is_array( $transformations ) ) {
+			$transformations = Api::generate_transformation_string( $transformations, $relationship->asset_type );
```

Comment:
@DavidCramer 

I'm not sure we want to do this.

@Vdeub-cloudinary

With this suggested change, we'll override everything currently in the relationship table with the transformations within the post meta. This might revert changes done manually in the relationship table.

Do you have any concerns here?

Reply:
we really do. 
the relationship table its stored ad a string, in meta its stored as an array.
In the the get transformations method, i've added the flag to deliver in a string or an array.